autogen/agentchat/contrib/retrieve_user_proxy_agent.py

```diff
@@ -239,17 +275,92 @@ def retrieve_docs(self, problem: str, n_results: int = 20, search_string: str =
         self._collection = True if self._docs_path is None else False  # whether the collection is created
         self._ipython = get_ipython()
         self._doc_idx = -1  # the index of the current used doc
-        self._results = {}  # the results of the current query
+        self._results = []  # the results of the current query
         self._intermediate_answers = set()  # the intermediate answers
         self._doc_contents = []  # the contents of the current used doc
         self._doc_ids = []  # the ids of the current used doc
         self._search_string = ""  # the search string used in the current query
+        self._distance_threshold = self._retrieve_config.get("distance_threshold", -1)
         # update the termination message function
         self._is_termination_msg = (
             self._is_termination_msg_retrievechat if is_termination_msg is None else is_termination_msg
         )
+        if isinstance(self._vector_db, str):
+            self._vector_db = VectorDBFactory.create_vector_db(
+                self._vector_db, path="tmp/db", embedding_function=self._embedding_function
+            )
```

Comment:
Should I create another variable on retrieve_user_proxy_agent called "vectordb_connection_string"?

```python
self._vector_db_connection_string="postgresql://username:password@hostname:port/database"
```

I would pass that into the second if statement as the connection_string parameter:

```suggestion
if isinstance(self._vector_db, str) and self._vector_db == "chroma":
   self._vector_db = VectorDBFactory.create_vector_db(
      db_name=self._vector_db, path="tmp/db", embedding_function=self._embedding_function
   )
elif isinstance(self._vector_db, str) and self._vector_db == "pgvector": 
   self._vector_db = VectorDBFactory.create_vector_db(
      db_name=self._vector_db, connection_string=_vector_db_connection_string, 
      embedding_function=self._embedding_function
   )   
```